tensorflow/lite/micro/examples/hello_world/train.py

```diff
@@ -0,0 +1,156 @@
+  y_values = np.sin(x_values).astype(np.float32)
+
+  # We'll use 60% of our data for training
+  train_split = int(0.6 * 1000)
+  test_split = int(0.2 * 1000 + train_split)
+
+  # Use np.split to chop our data into three parts.
+  # The second argument to np.split is an array of indices where the data will
+  # be split. We provide two indices, so the data will be divided into three
+  # chunks.
+  splits = np.split(x_values, [train_split, test_split])
+  x_train, x_validate = splits[0], splits[2]
+  splits = np.split(y_values, [train_split, test_split])
+  y_train, y_validate = splits[0], splits[2]
+
+  return (x_train, y_train, x_validate, y_validate)
+
+
+def create_model() -> tf.keras.Model:
+  model = tf.keras.Sequential()
+
+  # First layer takes a scalar input and feeds it through 16 "neurons". The
+  # neurons decide whether to activate based on the 'relu' activation function.
+  model.add(tf.keras.layers.Dense(16, activation='relu', input_shape=(1, )))
+
+  # The new second and third layer will help the network learn more complex
+  # representations
+  model.add(tf.keras.layers.Dense(16, activation='relu'))
+
+  # Final layer is a single neuron, since we want to output a single value
+  model.add(tf.keras.layers.Dense(1))
+
+  # Compile the model using the standard 'adam' optimizer and the mean squared
+  # error or 'mse' loss function for regression.
+  model.compile(optimizer='adam', loss='mse', metrics=['mae'])
+
+  return model
+
+
+def convert_tflite_model(model):
+  """Convert the save TF model to tflite model, then save it as .tflite flatbuffer format
+    Args:
+        model (tf.keras.Model): the trained hello_world Model
+    Returns:
+        The converted model in serialized format.
+  """
+  converter = tf.lite.TFLiteConverter.from_keras_model(model)
+  tflite_model = converter.convert()
+  return tflite_model
+
+
+def save_tflite_model(tflite_model, save_dir, model_name):
+  """save the converted tflite model
+  Args:
+      tflite_model (binary): the converted model in serialized format.
+      save_dir (str): the save directory
+      model_name (str): model name to be saved
+  """
+  if not os.path.exists(save_dir):
+    os.makedirs(save_dir)
+  save_path = os.path.join(save_dir, model_name)
+  with open(save_path, "wb") as f:
+    f.write(tflite_model)
+  logging.info("Tflite model saved to %s", save_dir)
+
+
+def train_model(epochs, x_train, y_train, x_validate, y_validate):
+  """Train keras hello_world model
+    Args: epochs (int) : number of epochs to train the model
+        x_train (numpy.array): list of the training data
+        y_train (numpy.array): list of the corresponding array
+    Returns:
+        tf.keras.Model: A trained keras hello_world model
+  """
+  model = create_model()
+  model.fit(x_train,
+            y_train,
+            epochs=epochs,
+            validation_split=0.2,
+            batch_size=64,
+            validation_data=(x_validate, y_validate),
+            verbose=2)
```

Comment:
can we use model.fit to automatically do the validation_split similar to:
https://github.com/tensorflow/tflite-micro/blob/ecdfe8fb324e4b4049dfd5d962b9d7c724f39a86/tensorflow/lite/micro/examples/mnist_lstm/train.py#L95-L100

Reply:
done